### DISCUSSION TITLE
Prefactors in scalar products

### DISCUSSION THREAD

**Discussion by mhutt21** [FeynCalc/feyncalc#173] [Q&A] [answered]
Hi, I am having a (perhaps quite basic) problem with expanding scalar products. Essentially I have a large expression which contains many scalar products `SP[P,Q]` and then I have a replacement rule for one of the momenta in each scalar product. The replacement rule is of the form `P -> a Momentum[P1] + b Momentum[P2]` where `a` and `b` are constants (they are actually ratios of other scalar products). Is there a way that I can tell FeynCalc that `a` and `b` are not part of the momenta, so they can be taken out of the scalar product?

i.e. the following
`product = SP[P,Q]`
`Pnew = a Momentum[P1] + b Momentum[P2]`
`product /. {P -> Pnew} // ExpandScalarProduct`
yields `Q . (a P1) + Q . (b P2)` but I need it to give `a (Q . P1) + b (Q . P2)` since `a` and `b` are just constants. Is there a way to do this?

Thanks in advance!

**Comment by vsht** [comment, marked as answer]
Hi,

for that you need to use `FCVariable`

https://feyncalc.github.io/FeynCalcBookDev/FCVariable.html

```
DataType[a, FCVariable] = True;
DataType[b, FCVariable] = True;

ExpandScalarProduct[SP[P,Q]/.P->a P1 +b P2]
StandardForm[%]
(*a Pair[Momentum[P1], Momentum[Q]] + b Pair[Momentum[P2], Momentum[Q]]*)
```

Cheers,
Vladyslav
